lib/commons/math/get-offset.js

```diff
@@ -1,38 +1,52 @@
-import { getTargetSize } from '../dom';
+import { getTargetRects } from '../dom';
+import { getBoundingRect } from './get-bounding-rect';
+import { isPointInRect } from './is-point-in-rect';
+import { getRectCenter } from './get-rect-center';
+import rectHasMinimumSize from './rect-has-minimum-size';
 
 /**
- * Get the offset between node A and node B
+ * Get the offset between target A and neighbor B. This assumes that the target is undersized and needs to check the spacing exception.
  * @method getOffset
  * @memberof axe.commons.math
- * @param {VirtualNode} vNodeA
- * @param {VirtualNode} vNodeB
+ * @param {VirtualNode} vTarget
+ * @param {VirtualNode} vNeighbor
  * @param {Number} radius
  * @returns {number}
  */
-export default function getOffset(vNodeA, vNodeB, minRadiusNeighbour = 12) {
-  const rectA = getTargetSize(vNodeA);
-  const rectB = getTargetSize(vNodeB);
+export default function getOffset(vTarget, vNeighbor, minRadiusNeighbour = 12) {
+  const targetRects = getTargetRects(vTarget);
+  const neighborRects = getTargetRects(vNeighbor);
 
-  // one of the rects is fully obscured
-  if (rectA === null || rectB === null) {
+  if (!targetRects.length || !neighborRects.length) {
     return 0;
   }
 
-  const centerA = {
-    x: rectA.x + rectA.width / 2,
-    y: rectA.y + rectA.height / 2
-  };
-  const centerB = {
-    x: rectB.x + rectB.width / 2,
-    y: rectB.y + rectB.height / 2
-  };
-  const sideB = getClosestPoint(centerA, rectB);
+  const targetBoundingBox = targetRects.reduce(getBoundingRect);
+  const targetCenter = getRectCenter(targetBoundingBox);
 
-  return Math.min(
-    // subtract the radius of the circle from the distance
-    pointDistance(centerA, centerB) - minRadiusNeighbour,
-    pointDistance(centerA, sideB)
-  );
+  // calculate distance to the closest edge of each neighbor rect
+  let minDistance = Infinity;
+  for (const rect of neighborRects) {
+    if (isPointInRect(targetCenter, rect)) {
+      return 0;
+    }
+
+    const closestPoint = getClosestPoint(targetCenter, rect);
+    const distance = pointDistance(targetCenter, closestPoint);
+    minDistance = Math.min(minDistance, distance);
+  }
+
+  if (rectHasMinimumSize(minRadiusNeighbour * 2, targetBoundingBox)) {
```

Comment:
This should be checking against the target-size of the neighbor, not the bounding box size of the target

Reply:
Good catch

Reply:
Also maybe good to leave a comment explaining the motivation for the early exit path

```suggestion
  // This early exit accounts for the final "undersized" qualifier in SC 2.5.8's spacing exception,
  // "...the circles do not intersect another target or the circle for another *undersized* target"
  if (rectHasMinimumSize(minRadiusNeighbour * 2, neighborTargetSize)) {
```